scheduler/src/test/java/gov/nasa/jpl/aerie/scheduler/UncontrollableDurationTest.java

```diff
@@ -132,12 +134,44 @@ public void testTimeDependent(){
+    assertTrue(TestUtility.containsActivity(plan, planningHorizon.fromStart("PT0S"), planningHorizon.fromStart("PT0S"), problem.getActivityType("SolarPanelNonLinearTimeDependent")));
+    assertTrue(TestUtility.containsActivity(plan, planningHorizon.fromStart("PT11M57S"), planningHorizon.fromStart("PT16M40S"), problem.getActivityType("SolarPanelNonLinearTimeDependent")));
+    assertTrue(TestUtility.containsActivity(plan, planningHorizon.fromStart("PT28M34S"), planningHorizon.fromStart("PT33M20S"), problem.getActivityType("SolarPanelNonLinearTimeDependent")));
+    assertTrue(TestUtility.containsActivity(plan, planningHorizon.fromStart("PT33M20S"), planningHorizon.fromStart("PT36M47S"), problem.getActivityType("SolarPanelNonLinearTimeDependent")));
+    assertTrue(TestUtility.containsActivity(plan, planningHorizon.fromStart("PT0S"), planningHorizon.fromStart("PT2M21S"), problem.getActivityType("SolarPanelNonLinearTimeDependent")));
+    assertTrue(TestUtility.containsActivity(plan, planningHorizon.fromStart("PT16M40S"), planningHorizon.fromStart("PT17M18S"), problem.getActivityType("SolarPanelNonLinearTimeDependent")));
   }
 
+  @Test
+  public void testBug(){
+    final var solarPanelActivityTrapezoidal = new ActivityInstance(problem.getActivityType("ControllableDurationActivity"),
+                                                                   Duration.of(1, Duration.MICROSECONDS),
+                                                                   Duration.of(3, Duration.MICROSECONDS));
+
+    final var solarPanelActivityTriangle = new ActivityCreationTemplate.Builder()
+        .ofType(problem.getActivityType("ZeroDurationUncontrollableActivity"))
+        .withTimingPrecision(Duration.of(1, Duration.MICROSECONDS))
+        .build();
+
+    //this time expression produces an interval [TimeAnchor.END, TimeAnchor.END + 2 Ms]
+    final var intervalStartTimeExpression = new TimeExpressionRelativeFixed(TimeAnchor.END, false);
+    intervalStartTimeExpression.addOperation(TimeUtility.Operator.PLUS, Duration.of(2, Duration.MICROSECONDS));
+
+    final var coexistenceTriangle = new CoexistenceGoal.Builder()
+        .thereExistsOne(solarPanelActivityTriangle)
+        .forAllTimeIn(planningHorizon.getHor())
+        .forEach(ActivityExpression.ofType(problem.getActivityType("ControllableDurationActivity")))
+        .startsAt(intervalStartTimeExpression)
+        .build();
+
+    problem.setGoals(List.of(coexistenceTriangle));
+    final var initialPlan = new PlanInMemory();
+    initialPlan.add(solarPanelActivityTrapezoidal);
+    problem.setInitialPlan(initialPlan);
+
+    final var solver = new PrioritySolver(problem);
+    final var plan = solver.getNextSolution().get();
+    solver.printEvaluation();
+    assertTrue(TestUtility.containsActivity(plan, planningHorizon.fromStart("PT0.000004S"), planningHorizon.fromStart("PT0.000004S"), problem.getActivityType("ZeroDurationUncontrollableActivity")));
+  }
```

Comment:
The bug happened in the situation in which there was no end timing constraint but an interval start constraint in addition to a 0-duration uncontrollable activity.

Reply:
So prior to this PR `ZeroDurationUncontrollableActivity` would not be placed as the first planned activity?

Reply:
Correct, @pcrosemurgy - the issue that came up was if you have a scheduling goal that just says "Place Activity B any time after the end of Activity A" - this algorithm would throw a `DivergenceException` (which would result in the activity not being placed).

The issue comes up when the start interval is much more constrained than the end interval. The algorithm tries to find a solution where the activity ends within a certain tolerance of the midpoint of the end interval. If the end interval is very large, there may not be any solution that respects the start interval...

(Now that I've typed that out, I think this algorithm will still break if for some reason neither of the first two endpoints is a valid solution 🤔 but maybe that case will never happen? I'll need to ponder...)

Reply:
I'll try building a test case with this case in mind. Maybe it'll lead to another PR :D